addons/ShaderFunction-Extras/Shapes/shapes.gdshaderinc

```diff
@@ -0,0 +1,59 @@
+vec4 polygon(vec2 uv, float width, int sides) {
+	uv = uv * 2.0 - 1.0;
+	float angle = atan(uv.x, uv.y);
+	float radius = 6.28318530718 / float(sides);
+	float dist = cos(floor(0.5 + angle / radius) * radius - angle) * length(uv);
+	float poly = step(width, dist);
```

Comment:
You can also just leave the poly as a `float` output same as the other shape functions since they are just SDFs

Reply:
no, instead just return the `poly` or other named shape variables before you put them in a `vec4` after that